lib/style-transform.js

```diff
@@ -1,156 +1,82 @@
 const stylis = require('stylis')
 
-/**
- * @type {RegExp}
- *
- * matches :global($1) capturing $1
- */
-const regExpReplaceGlobal = /:global\((.*)\)/g
+function disableNestingPlugin() {
+  const context = arguments[0]
+  if (context === 2) {
+    const parent = (arguments[3] || [])[0]
+    if (typeof parent === 'string' && parent.trim().length > 0) {
+      throw new Error('Unfortunately nesting is not supported by styled-jsx.')
```

Comment:
@giuseppeg Optionally this could also print line and column number, so you can get to the origin of the error in the source file.